### PR TITLE
[#1338] feat(jdbc): Postgresql supports auto_increment.

### PR DESCRIPTION
**Pull request by Clearvive**
### What changes were proposed in this pull request?
Postgresql supports auto_increment.

### Why are the changes needed?
Fix: #1338

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
UT

### CODE REVIEW

File: catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlTableOperations.java [hunk 1]
```diff
@@ -284,6 +285,13 @@ private StringBuilder appendColumnDefinition(JdbcColumn column, StringBuilder sq
         .append(typeConverter.fromGravitinoType(column.dataType()))
         .append(SPACE);
 
+    if (column.autoIncrement()) {
+      if (!Types.IntegerType.get().equals(column.dataType())) {
```
Comment (yuqi1129) [inline catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlTableOperations.java:289]: What about type `BIGSERIAL`?

**Comment by mchades** [comment]
Maybe the issues of IT and code conflicts should be resolved first before review

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/postgresql/TestPostgreSqlTableOperations.java [hunk 2]
```diff
@@ -152,18 +155,16 @@ public void testOperationTable() {
...
         new JdbcColumn.Builder()
             .withName("col_1_new")
-            .withType(VARCHAR)
+            .withType(Types.LongType.get())
             .withComment("test_new_comment")
             .withNullable(false)
+            .withAutoIncrement(true)
             .build());
     alterColumns.add(
         new JdbcColumn.Builder()
```
Comment (jerryshao) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/postgresql/TestPostgreSqlTableOperations.java:170]: Can you please add tests verify that

1. auto increment is successfully set in the underlying DB.
2. auto increment can be gotten correctly from the underlying DB.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/postgresql/TestPostgreSqlTableOperations.java [hunk 3]
```diff
@@ -398,4 +399,70 @@ public void testCreateMultipleTable() throws SQLException {
...
+        new JdbcColumn.Builder()
+            .withName("col_2")
+            .withType(INT)
+            .withNullable(false)
+            .withComment("set test key")
+            .build());
+    Assertions.assertThrows(
+        IllegalArgumentException.class,
+        () ->
+            TABLE_OPERATIONS.create(
```
Comment (jerryshao) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/postgresql/TestPostgreSqlTableOperations.java:460]: What's the exception here?

Comment (jerryshao) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/postgresql/TestPostgreSqlTableOperations.java:460]: Can you please a): and the comment to this test to clarify what you test; b) verify the exception message below.

File: catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlTableOperations.java [hunk 4]
```diff
@@ -284,6 +286,17 @@ private StringBuilder appendColumnDefinition(JdbcColumn column, StringBuilder sq
         .append(typeConverter.fromGravitinoType(column.dataType()))
         .append(SPACE);
 
+    if (column.autoIncrement()) {
+      if (!Types.allowAutoIncrement(column.dataType())) {
+        throw new IllegalArgumentException(
+            "Not support auto-increment , column: "
```
Comment (jerryshao) [inline catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlTableOperations.java:292]: Please change to: “Unsupported auto-increment, column: "

**Review by jerryshao — APPROVED** [review]

**Pull request merged**